.circleci/build_image.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+set -xe
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 DOCKERFILE IMAGE_NAME"
+  exit 2
+fi
+
+DIR=`dirname $1`
+DOCKERFILE=$1
+IMAGE_NAME=$2
+
+if [[ -z $CIRCLE_PULL_REQUEST ]]; then
+  # Log in to Docker Hub.
+  docker login -u $DOCKER_USER -p $DOCKER_PASS
+  # Decide whether Docker Hub images should be tagged ":branch-Y" or ":latest".
+  if [[ $CIRCLE_BRANCH != "master" ]]; then
```

Comment:
Nit: This `DOCKERHUB_TAG` variable is only used to tag the image after it was built. Please move this `if [[ $CIRCLE_BRANCH != "master" ]]` block into the `if [[ -z $CIRCLE_PULL_REQUEST ]]` at the bottom of this file.